Source/WebGPU/WebGPU/Sampler.h

```diff
@@ -87,13 +79,13 @@ class Sampler : public WGPUSamplerImpl, public RefCounted<Sampler> {
     const Ref<Device> m_device;
     // static is intentional here as the limit is per process
     static Lock samplerStateLock;
-    using CachedSamplerStateContainer = HashMap<GenericHashKey<UniqueSamplerIdentifier>, WeakObjCPtr<id<MTLSamplerState>>>;
+    using CachedSamplerStateContainer = HashMap<UniqueSamplerIdentifier, WeakObjCPtr<id<MTLSamplerState>>>;
     struct SamplerStateWithReferences {
         RetainPtr<id<MTLSamplerState>> samplerState;
-        HashSet<GenericHashKey<UniqueSamplerIdentifier>> apiSamplerList;
+        HashSet<const Sampler*> apiSamplerList;
     };
-    using RetainedSamplerStateContainer = HashMap<GenericHashKey<UniqueSamplerIdentifier>, SamplerStateWithReferences>;
```

Comment:
lines 95 and 96 were good improvements, I would keep these while re-landing the original PR